sqlmesh/core/linter/rule.py

```diff
@@ -22,6 +22,20 @@ class RuleLocation(PydanticModel):
     start_line: t.Optional[int] = None
 
 
+class Position(PydanticModel):
+    """The position of a rule violation in a file, the position follows the LSP standard."""
+
+    line: int
+    character: int
+
+
+class Range(PydanticModel):
+    """The range of a rule violation in a file. The range follows the LSP standard."""
+
+    start: Position
+    end: Position
```

Comment:
Should we document that this represents a `[start, end)` range?

Reply:
To me this is really what I was hoping 

`The range follows the LSP standard.`

^ I don't want to reinvent the wheel, I just want them to follow the LSP standard. So if you are confused, just go there.